test/ref/reduce_mean.cpp

```diff
@@ -109,3 +209,28 @@ TEST_CASE(reduce_mean_int)
     std::vector<int> gold{2, 6, 10};
     EXPECT(results_vector == gold);
 }
+
+TEST_CASE(reduce_mean_variable_axes12_int)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+
+    migraphx::shape x_shape{migraphx::shape::int32_type, {3, 2, 2}};
+    auto x = mm->add_parameter("x", x_shape);
+    migraphx::shape axes_shape{migraphx::shape::int64_type, {2}};
+    auto axes = mm->add_parameter("axes", axes_shape);
+    mm->add_instruction(migraphx::make_op("reduce_mean"), x, axes);
+    p.compile(migraphx::make_target("ref"));
+
+    migraphx::parameter_map pm;
+    std::vector<int32_t> x_arg{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};
+    pm["x"] = migraphx::argument(x_shape, x_arg.data());
+    std::vector<int64_t> axes_arg{1, 2};
+    pm["axes"]  = migraphx::argument(axes_shape, axes_arg.data());
+    auto result = p.eval(pm).back();
+    std::vector<int32_t> results_vector;
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+
+    std::vector<int32_t> gold{2, 6, 10};
+    EXPECT(results_vector == gold);
+}
```

Comment:
If there is not a test for default axes, with keepdims=1, the following (copied from Netron) would be a good test. Thanks.

```
shape = [3, 2, 2]
axes = np.array([], dtype=np.int64)
keepdims = 1

node = onnx.helper.make_node(
    "ReduceMean",
    inputs=["data", "axes"],
    outputs=["reduced"],
    keepdims=keepdims,
)

data = np.array(
    [[[5, 1], [20, 2]], [[30, 1], [40, 2]], [[55, 1], [60, 2]]],
    dtype=np.float32,
)
reduced = np.mean(data, axis=None, keepdims=keepdims == 1)
# print(reduced)
# [[[18.25]]]

expect(
    node,
    inputs=[data, axes],
    outputs=[reduced],
    name="test_reduce_mean_default_axes_keepdims_example",
)
```

Reply:
Please note, the dimension of the output tensor has to be verified as well as the value.